### ISSUE TITLE
Fetch error

### ISSUE THREAD

**Issue by NineXYZ** [sudipghimire533/ytui-music#43]
The music section does not load at all, no matter what artist name I put in the search bar.
I guess it's a problem with the invidious instance so I tried a few random ones and that didn't work either.
![image](https://github.com/sudipghimire533/ytui-music/assets/66285369/4425eb0e-49e2-426e-a16d-34a603fcf585)

**Comment by AxiomOfChoices** [comment]
+1 On the issue, may I ask what desktop environment setup you are using? It looks so nice.

**Comment by NotoriousArnav** [comment]
+1 on the issue, and I think its Hyprland

**Comment by NineXYZ** [comment]
Correct, it's Hyprland. I changed my system since that screenshot, so I don't have the dotfiles anymore.

**Comment by yugonline** [comment]
+1 on the issue. Nice rice. I cannot play anything. If I play a playlist it just keep skipping tracks and then reaches the end of playlist and if I am playing a track it again keeps trying different URLs and then either a Fetch Error or worse, no result.

**Comment by 3isvogel** [comment]
I get the same error with this specific search, other strings work just fine, youtube-dl has a similar error when fetching for login-protected content (For example "Yoru ni kakeru" from Yoasobi, since it has sensible content, is age-restricted and accessible only when logged in) the only solution I found to avoid this is to login from your browser, dump your session cookies in a file and provide that cookie file to youtube-dl (or yt-dlp if you are using that one instead) I don't know any other way to provide youtube-dl your login info, I will try and tell you if I can get it to work

**Comment by MasterRoby3** [comment]
+1 on the issue. Just impossible to fetch anything (i get fetch error on all categories after changing the invidious instance)

**Comment by hubyhuby** [comment]
I had fetch error before installing : "sudo apt install -y youtube-dl libmpv1 libmpv-dev
"

**Comment by Grandkahuna43325** [comment]
The problem is with the invidious instances, some of them went down because of... well us, too many projects point to instances and they close. 
I can only recommend changing instances in config(~/.config/ytui_music/config.json line 73).
You can find servers on https://docs.invidious.io/instances/ 

The app doesn't indicate why fetch error occurred, I'll probably fork it and play around when I have some time(so probably never) for now the solution is(probably) changing server list

**Comment by CarterSnich** [comment]
> The problem is with the invidious instances, some of them went down because of... well us, too many projects point to instances and they close. I can only recommend changing instances in config(~/.config/ytui_music/config.json line 73). You can find servers on https://docs.invidious.io/instances/
> 
> The app doesn't indicate why fetch error occurred, I'll probably fork it and play around when I have some time(so probably never) for now the solution is(probably) changing server list
> 
> PS: if you have a title with space, change the space to _

Is it possible to set Youtube Music as server? I really thought this gets music directly from yt music.

**Comment by foxyseta** [comment]
@Grandkahuna43325 Yeah, logging would be nice.

**Comment by Grandkahuna43325** [comment]
> Is it possible to set Youtube Music as server? I really thought this gets music directly from yt music.

Nope

**Comment by shuntia** [comment]
> > Is it possible to set Youtube Music as server? I really thought this gets music directly from yt music.
> 
> Nope

You *technically* can do this by using an `<iframe>` for playback (similar to how `yt-dlp` works) and the `YouTube Data API v3`. However, this approach would require significant rewriting, and you'd also need to include a mandatory local API key in your configs. This setup could be cumbersome to manage.

That said, while it would involve a performance tradeoff, it might offer more stability and be a bit more legally compliant than other methods.

**Comment by Sai-Prabhav** [comment]
+1 same fetch error and I am not able to play any song 

![Image](https://github.com/user-attachments/assets/55d7af5f-68f3-486f-97ff-05ec656c3b59)

**Comment by Grandkahuna43325** [comment]
> +1 same fetch error and I am not able to play any song
> 

as mentioned above, you can change instances in config

**Comment by rafapgomes** [comment]
> > +1 same fetch error and I am not able to play any song
> 
> as mentioned above, you can change instances in config

sorry if its a dumb question but how i see instances that work? i add yewtube i think but still not works..

**Comment by Grandkahuna43325** [comment]
> sorry if its a dumb question but how i see instances that work? i add yewtube i think but still not works..

Visit https://docs.invidious.io/instances/ pick one and ping it or visit their website

**Comment by gasad-dev** [comment]
+1 cause don't work any solution here(

**Comment by sudipghimire533** [comment]
@gasad-dev  Can you make sur the invidious instance you are pointing to is exposing API

**Comment by gasad-dev** [comment]
> [@gasad-dev](https://github.com/gasad-dev) Can you make sur the invidious instance you are pointing to is exposing API

Yes, I'm sure of it, but it still writes an error
And can you provide a config that works well? Because I could have made a mistake when fixing the config

**Comment by capy-yy** [comment]
+1 happens on every instance I've tried

**Comment by usernameisunavailable-exe** [comment]
+1 on the issue. I don't have mutch to say...
